libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxAudioGuidance.kt

```diff
@@ -160,15 +163,25 @@ internal constructor(
         }
     }
 
-    private fun MapboxNavigation.audioGuidanceVoice(): Flow<MapboxAudioGuidanceVoice> =
-        combine(
+    @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+    private fun MapboxNavigation.audioGuidanceVoice(): Flow<MapboxAudioGuidanceVoice> {
+        var trigger: VoiceInstructionsPrefetcher? = null
+        return combine(
             mapboxVoiceInstructions.voiceLanguage(),
             configOwner!!.language(),
         ) { voiceLanguage, deviceLanguage -> voiceLanguage ?: deviceLanguage }
             .distinctUntilChanged()
             .map { language ->
-                audioGuidanceServices.mapboxAudioGuidanceVoice(this, language)
+                audioGuidanceVoice?.destroy()
+                trigger?.onDetached(this)
```

Comment:
As I see you call `trigger?.onDetached(this)` every time language changes. But how is it detached when subscription is cancelled?

Reply:
Did you mean that there will be n+1 `onAttached` invocations and n `onDetached` with the current approach? You're probably right, I added an `onDetached` invocation to `MapboxAudioGuidance#onDetached`.
And it made me realize that we don't have to both destroy audioGuidanceVoice and trigger. One invocation is enough.